test/e2e/test_create_and_xmlupload.py

```diff
@@ -0,0 +1,129 @@
+def test_project(auth_header: dict[str, str]) -> None:
+    get_project_url = f"{CREDS.server}/admin/projects/shortcode/{PROJECT_SHORTCODE}"
+    project = requests.get(get_project_url, headers=auth_header, timeout=3).json()["project"]
+    _check_project(project)
+
+    onto = requests.get(ONTO_IRI, headers=auth_header, timeout=3).json()["@graph"]
+    _check_props([elem for elem in onto if elem.get("knora-api:isResourceProperty")])
+    _check_resclasses([elem for elem in onto if elem.get("knora-api:isResourceClass")])
+
+
+def _check_project(project: dict[str, Any]) -> None:
+    assert project["shortname"] == "e2e-tp"
+    assert project["shortcode"] == PROJECT_SHORTCODE
+    assert project["longname"] == "e2e test project"
+    assert project["description"] == [{"value": "The e2e test project", "language": "en"}]
+    assert project["keywords"] == ["e2e-test-projec-keyword"]
+    assert project["ontologies"] == [ONTO_IRI]
+
+
+def _check_props(props: list[dict[str, Any]]) -> None:
+    assert len(props) == PROPS_IN_ONTO_JSON
+    assert props[0]["@id"] == f"{ONTO_NAME}:hasText"
+    assert props[0]["rdfs:subPropertyOf"] == {"@id": "knora-api:hasValue"}
+    assert props[0]["knora-api:objectType"] == {"@id": "knora-api:TextValue"}
+    assert props[0]["rdfs:label"] == "Text"
+
+
+def _check_resclasses(resclasses: list[dict[str, Any]]) -> None:
+    assert len(resclasses) == RESCLASSES_IN_ONTO_JSON
+    res_1, res_2 = resclasses
+
+    assert res_1["@id"] == f"{ONTO_NAME}:ImageResource"
+    assert res_1["rdfs:label"] == "Image Resource"
+    cards_1 = res_1["rdfs:subClassOf"]
+    fileval_card = [x for x in cards_1 if x.get("owl:onProperty", {}).get("@id") == "knora-api:hasStillImageFileValue"]
+    assert len(fileval_card) == 1
+    assert fileval_card[0].get("owl:cardinality") == 1
+    hasText_card = [x for x in cards_1 if x.get("owl:onProperty", {}).get("@id") == f"{ONTO_NAME}:hasText"]
+    assert len(hasText_card) == 1
+
+    assert res_2["@id"] == f"{ONTO_NAME}:PDFResource"
+    assert res_2["rdfs:label"] == "PDF Resource"
+
+
+@pytest.mark.usefixtures("_xmlupload")
+def test_xmlupload(auth_header: dict[str, str], project_iri: str) -> None:
+    img_resources = _get_resources(f"{ONTO_IRI}#ImageResource", auth_header, project_iri)
+    _analyze_img_resources(img_resources)
+    pdf_resources = _get_resources(f"{ONTO_IRI}#PDFResource", auth_header, project_iri)
+    _analyze_pdf_resources(pdf_resources)
+
+
+def _get_resources(resclass_iri: str, auth_header: dict[str, str], project_iri: str) -> list[dict[str, Any]]:
+    resclass_iri_encoded = urllib.parse.quote_plus(resclass_iri)
+    get_resources_route = f"{CREDS.server}/v2/resources?resourceClass={resclass_iri_encoded}&page=0"
+    headers = auth_header | {"X-Knora-Accept-Project": project_iri}
+    response = requests.get(get_resources_route, timeout=3, headers=headers).json()
+    resources: list[dict[str, Any]] = response.get("@graph", [response])
+    return resources
```

Comment:
just a detail: to me it seems a bit strange to have helper methods both before and after the test case(s). I would normally expect these "categories" of code to be grouped separately.

Reply:
Good point, I rearranged them.